server/store/store.go

```diff
@@ -58,3 +59,35 @@ func (s *Store) Delete(key string) error {
 	}
 	return nil
 }
+
+func (s *Store) DeleteUserOnSecretChange() error {
+	page := 0
+	for {
+		kvList, err := s.api.KVList(page, constants.UsersPerPage)
+		if err != nil {
+			return err
+		}
+		if len(kvList) == 0 {
+			return nil
+		}
+
+		isUserDeleted := false
```

Comment:
please explain the need for this flag and why we are updating the pagnumber according to this flag